src/rpcblockchain.cpp

```diff
@@ -1081,7 +1081,16 @@ UniValue advertisebeacon(const UniValue& params, bool fHelp)
     else
     {
         res.pushKV("Public Key",sOutPubKey.c_str());
-        res.pushKV("Warning!","Your keys to research rewards have been stored in your wallet. It is recommended that you back up your files on a regular basis.");
+
+        LOCK2(cs_main, pwalletMain->cs_wallet);
+
+        res.pushKV("Warning!","Your keys to research rewards have been stored in your wallet. A wallet and config file backup has automatically been kicked off.");
+
+        bool bWalletBackupResults = BackupWallet(*pwalletMain, GetBackupFilename("wallet.dat"));
+        bool bConfigBackupResults = BackupConfigFile(GetBackupFilename("gridcoinresearch.conf"));
```

Comment:
In AdvertiseBeacon, AdvertiseBeacon is already called. Considering that config backup is unnecessary, one backup is enough.

Reply:
Looks like the config backup is called in AdvertiseBeacon, and the actual sending of the contract is stopped if the backup fails. I think you are right, I should only backup the wallet here. I am ok with having it here rather than in the AdvertiseBeacon function.

Reply:
Although automatic advertisement probably calls AdvertiseBeacon. Let me go check that...

Reply:
the services loop in main.cpp on a TimerMain("send_beacon",180) calls AdvertiseBeacon. I think this means I should move the database backup there to to join the config backup.